xcode/Subconscious/Shared/Services/DatabaseService.swift

```diff
@@ -1062,17 +1097,18 @@ final class DatabaseService {
         )
         .compactMap({ row in
             guard
-                let address = row.col(0)?
+                let did = row.col(0)?.toString()?.toDid(),
+                let address = row.col(1)?
                     .toString()?
-                    .toLink()?
```

Comment:
Why was this step removed?

Reply:
Accidental, restored it, but I'm realising I don't fully grasp the importance of this step.

To my eye, the "benefit" of going via `toLink()` first is that we're asserting that there's a DID present in the address, going straight to `toSlashlink()` would allow a peerless address to pass through... but is that actually the job of this code? 

All entries in the `memo` table _should_ have a wellformed `id` already (ideally unit tested on the write-side), and we're throwing away the additional constraint by converting to a `Slashlink` right afterwards.

Maybe the core of my confusion comes down to this: when should we prefer to select `id` vs `slashlink` from the `memo` table if we're ultimately converting _either_ to a `Slashlink` before returning? My guess is this comes down to how `Did.local` is handled somehow, but now that we have `EntryStub.did` it would probably be better to consult _that_ to determine if content is local? It feels brittle to me to be implicitly depending on a particular `Peer` case within a `Slashlink`...